crates/scheduler/src/utils.rs

```diff
@@ -356,28 +350,23 @@ pub fn get_outgoing_webhook_retry_schedule_time(
     if retry_count == 0 {
         Some(retry_mapping.start_after)
     } else {
-        get_delay(
-            retry_count,
-            retry_mapping
-                .count
-                .iter()
-                .zip(retry_mapping.frequency.iter()),
-        )
+        get_delay(retry_count, &retry_mapping.frequencies)
     }
 }
 
 /// Get the delay based on the retry count
-fn get_delay<'a>(retry_count: i32, array: impl Iterator<Item = (&'a i32, &'a i32)>) -> Option<i32> {
+fn get_delay(retry_count: i32, vec: &[(i32, i32)]) -> Option<i32> {
     // Preferably, fix this by using unsigned ints
```

Comment:
Well, this is another thing that we should address. Can be in a separate PR, not a problem.